### PR TITLE
Add doc requirements to PR template

### PR DESCRIPTION
**Pull request by meker12**
# Description

Add release note and documentation requirements to PR template

### Fixed Issues (if relevant)

n/a

### Manual testing scenarios
n/a

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [ ] Pull request introduces user-facing changes and includes meaningful release notes and documentation
 - [ ] All commits are accompanied by meaningful commit messages

### CODE REVIEW

**Review by shiftedreality — DISMISSED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -16,10 +16,10 @@
 
 ### Fixed Issues (if relevant)
 <!---
-    If relevant, please provide a list of fixed issues in the format magento/magento-cloud-patches#<issue_number>.
+    If relevant, please provide a list of fixed issues in the format magento/magento-cloud-components#<issue_number>.
     There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
 -->
-1. magento/magento-cloud-patches#<issue_number>: Issue title
+1. magento/magento-cloud-components#<issue_number>: Issue title
```
Comment (andriyShevtsov) [inline .github/PULL_REQUEST_TEMPLATE.md]: Shouldn't it be patches if template  for patches?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 ### Fixed Issues (if relevant)
 <!---
-    If relevant, please provide a list of fixed issues in the format magento/magento-cloud-patches#<issue_number>.
```
Comment (meker12) [inline .github/PULL_REQUEST_TEMPLATE.md]: @shiftedreality  This is the Patches repo, so I think patches is correct here.  

FYI -- I added the PR template updates to ece-tools, patches, and components to facilitate dev work for release note and doc updates.

Comment (shiftedreality) [reply .github/PULL_REQUEST_TEMPLATE.md]: yeap, haven't notice that

**Review by shiftedreality — DISMISSED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -30,8 +30,17 @@
...
+
+### Associated documentation updates
+<!--
+ If your proposed update requires user documentation, submit a PR to the Magento DevDocs repository. For extensive updates requiring assistance, submit an issue to DevDocs. See https://github.com/magento/devdocs/blob/master/.github/CONTRIBUTING.md.
+ -->
+Add link to Magento DevDocs PR or Issue, if needed.
+
 ### Contribution checklist
  - [ ] Pull request has a meaningful description of its purpose
+ - [ ] Pull request introduces user-facing changes and includes meaningful release notes and documentation
```
Comment (meker12) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
 - [ ] Pull request introduces user-facing changes and includes meaningful updates for any required release notes and documentation changes
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -30,8 +30,17 @@
...
+<!--
+ If your proposed update requires user documentation, submit a PR to the Magento DevDocs repository. For extensive updates requiring assistance, submit an issue to DevDocs. See https://github.com/magento/devdocs/blob/master/.github/CONTRIBUTING.md.
+ -->
+Add link to Magento DevDocs PR or Issue, if needed.
+
 ### Contribution checklist
  - [ ] Pull request has a meaningful description of its purpose
+ - [ ] Pull request introduces user-facing changes and includes meaningful updates for any required release notes and documentation changes
  - [ ] All commits are accompanied by meaningful commit messages
- - [ ] All new or changed code is covered with unit/integration tests (if applicable)
```
Comment (shiftedreality) [inline .github/PULL_REQUEST_TEMPLATE.md:36]: Please revert this

**Review by shiftedreality — CHANGES_REQUESTED** [review]

**Review by shiftedreality — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (andriyShevtsov) [reply .github/PULL_REQUEST_TEMPLATE.md]: @meker12 ?

**Review by andriyShevtsov — APPROVED** [review]

**Review by shiftedreality — APPROVED** [review]

**Pull request merged**